staging/src/k8s.io/apiserver/pkg/storage/cacher.go

```diff
@@ -369,6 +369,12 @@ func (c *Cacher) Get(ctx context.Context, key string, resourceVersion string, ob
 		return err
 	}
 
+	if !c.ready.check() && getRV == 0 {
```

Comment:
@smarterclayton just wanted to make sure this actually helps us... if the gets were blocking while the lock was already held, it wouldn't